pkg/migrations/op_create_table.go

```diff
@@ -201,6 +209,15 @@ func (o *OpCreateTable) updateSchema(s *schema.Schema) *schema.Schema {
 			}
 		case ConstraintTypePrimaryKey:
 			primaryKeys = c.Columns
+		case ConstraintTypeForeignKey:
+			foreignKeys[c.Name] = &schema.ForeignKey{
+				Name:              c.Name,
+				Columns:           c.Columns,
+				ReferencedTable:   c.References.Table,
+				ReferencedColumns: c.References.Columns,
```

Comment:
couldn't figure out why we ignore match type here and also when generating constraint sql, in `WriteForeignKey`. 
Though, we take it into account by setting the match type field in `convertConstraint`

Couldn't see any comments about it. Could you please explain it a bit?

Reply:
No reason. I did not add it because it was a new attribute, and I planned to do unifying the existing `ForeignKey` types in a later PR. I can add it here, if that makes things clearer for you.

Reply:
Updated.